### PR TITLE
Update Andorra parish holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update holidays of Canillo, Sant Julià de Lòria, Andorra la Vella.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by mborsetti** [comment]
Hmmm...I just noticed that for this country we use the "AD-02" etc notation for subdivisions.  For all other countries we're using the second part only, the first part being the [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) code of the country, e.g.:

```
calif_holidays = holidays.country_holidays(country='US', subdiv='CA')
```

These should be changed to the project's standards, i.e. the below should work:

```
canillo_holidays = holidays.country_holidays(country='AD', subdiv='02')
```

File: tests/countries/test_andorra.py [hunk 1]
```diff
@@ -104,46 +105,46 @@ def test_2023(self):
 
         # AD-02, Canillo.
         self.assertHoliday(
-            Andorra(subdiv="AD-02", years=2023),
-            "2023-07-21",
-            "2023-07-22",
-            "2023-07-23",
+            Andorra(subdiv="02", years=2023),
```
Comment (arkid15r) [inline tests/countries/test_andorra.py:108]: Just FYI:
```suggestion
            Andorra(subdiv="02"),
```
Would look cleaner (yeah, I know I wrote this couple months ago :) )

**Review by arkid15r — COMMENTED** [review]
👍

File: holidays/countries/andorra.py [hunk 2]
```diff
@@ -83,46 +85,30 @@ def _populate(self, year: int) -> None:
         self[date(year, DEC, 26)] = "Saint Stephen's Day"
 
         # Parish local holidays.
+        ad_02_dt = _get_nth_weekday_of_month(3, SAT, JUL, year)
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/andorra.py]: Done.

**Pull request merged**